fluent_pose_synthesis/network/models.py

```diff
@@ -1,85 +1,65 @@
-    """
-    def __init__(self, input_feats: int, latent_dim: int, keypoints: int, dims: int) -> None:
-        super().__init__()
-        self.output_process = OutputProcess(input_feats, latent_dim, keypoints, dims)
-    
     def forward(self, x: torch.Tensor) -> torch.Tensor:
-        # x: (B, T, latent_dim)
-        x = x.transpose(0, 1)           # Convert to (T, B, latent_dim)
-        out = self.output_process(x)    # Output shape: (B, keypoints, dims, T)
-        out = out.permute(0, 3, 1, 2)     # Convert to (B, T, keypoints, dims)
-        return out
-
+        if self.pre_transform is not None:
+            x = self.pre_transform(x)
+        else:
+            if x.dim() >= 3:
+                x = x.transpose(0, 1)  # Convert (B, T, ...) to (T, B, ...)
+        x = self.module(x)
+        if self.post_transform is not None:
+            x = self.post_transform(x)
+        else:
+            if x.dim() >= 3:
+                x = x.transpose(0, 1)  # Convert back from (T, B, ...) to (B, T, ...)
+        return x
 
-class BatchFirstMotionProcess(nn.Module):
-    """
-    Wrapper for CAMDM's MotionProcess to support batch-first input."""
-    def __init__(self, input_feats: int, latent_dim: int) -> None:
-        super().__init__()
-        self.motion_process = MotionProcess(input_feats, latent_dim)
-    
-    def forward(self, x: torch.Tensor) -> torch.Tensor:
-        # x: (B, T, keypoints, dims)
-        B, T, keypoints, dims = x.shape
-        # Convert to (B, keypoints, dims, T)
-        x = x.permute(0, 2, 3, 1)
-        # Call CAMDM's MotionProcess which returns shape (T, B, latent_dim)
-        out = self.motion_process(x)
-        out = out.transpose(0, 1)     # Convert to (B, T, latent_dim)
-        return out
+    def __getattr__(self, name: str):
+        try:
+            return super().__getattr__(name)
+        except AttributeError:
+            return getattr(self.module, name)
 
 
 class DisfluentContextEncoder(nn.Module):
     """
-    Encodes the disfluent sequence to a global context vector using a Transformer and mean pooling.
-    Input shape: (B, T, input_feats)
+    Encodes the disfluent sequence into a global context vector using a Transformer encoder and mean pooling.
+    Expected input shape: (B, T, keypoints, dims)
     """
-    def __init__(self, input_feats: int, latent_dim: int, num_layers: int = 2, num_heads: int = 4, dropout: float = 0.1) -> None:
+    def __init__(self, input_feats: int, latent_dim: int, num_layers: int = 2, num_heads: int = 4, dropout: float = 0.1):
         super().__init__()
-        self.pose_encoder = BatchFirstMotionProcess(input_feats, latent_dim)
+        # Use MotionProcess with custom dimension transforms:
+        self.pose_encoder = BatchFirstWrapper(
+            MotionProcess(input_feats, latent_dim),
+            pre_transform=lambda x: x.permute(0, 2, 3, 1),  # (B, T, keypoints, dims) -> (B, keypoints, dims, T)
```

Comment:
interesting, i didn't realize every class would need a special transform, i thought it would always be `transpose(0,1)`
interesting to know why